lib/spack/docs/packaging_guide_creation.rst

```diff
@@ -777,28 +782,41 @@ The logic behind this sort order is two-fold:
 #. The most-recent development version of a package will usually be newer than any released numeric versions.
    This allows the ``@develop`` version to satisfy dependencies like ``depends_on(abc, when="@x.y.z:")``
 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Specifying versions in other directives
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Many Spack directives accept versions speciers, for example
+.. _version_constraints:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Specifying version constraints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Many Spack directives accept version constraints to restrict the spec to a given version using the ``@=<version>`` syntax, or a range using the ``@<specifier>`` syntax.
+Refer to :ref:`version-specifier` for the complete syntax.
+Version range constraints are useful for :ref:`version_compatibility`.
+
+For example,
 
 .. code-block:: python
 
-   depends_on("python@3")
-   conflicts("foo@1.2.3:", when="@:4.5")
+   depends_on("foo")
+   depends_on("python@=3.10.1")
```

Comment:
If we're discouraging this, I think it's better not to highlight it in the first example.